### PR TITLE
Refactor "applyLocale" and remove references to "language" column in UFMatch table

### PR DESCRIPTION
**Pull request by christianwach**
Overview
----------------------------------------
Fixes issues raised in [this ticket on Lab](https://lab.civicrm.org/dev/core/-/issues/1891).

Before
----------------------------------------
* `CRM_Core_BAO_ConfigSetting::applyLocale()` does not apply the desired locale when `inheritLocale` is specified.
* `CRM_Core_BAO_ConfigSetting::applyLocale()` reads and saves the locale in the UFMatch table with no discernible use case in doing so.
* `CRM_Admin_Form_Setting_Localization::onChangeLcMessages()` saves the locale in the UFMatch table for the contact/user that saves the setting, again with no discernible effect.

After
----------------------------------------
* `CRM_Core_BAO_ConfigSetting::applyLocale()` correctly applies the desired locale when `inheritLocale` is specified.
* `CRM_Core_BAO_ConfigSetting::applyLocale()` and `CRM_Admin_Form_Setting_Localization::onChangeLcMessages()` no longer touch the `language` column in the UFMatch table.

Technical Details
----------------------------------------
See [this ticket on Lab](https://lab.civicrm.org/dev/core/-/issues/1891) for full discussion.

tl;dr This PR fixes a regression where `CRM_Core_BAO_ConfigSetting::applyLocale()` no longer applies the desired locale when `inheritLocale` is specified and removes references to the seemingly-unused `language` column in the UFMatch table.

Comments
----------------------------------------
* I don't know how to create the upgrade script to remove the column from the table itself.
* I'm not sure how to write tests for this.

### CODE REVIEW

**Comment by mattwire** [comment]
I think we can leave the upgrader step for a later version of CiviCRM - in case there are any wierd and wonderful customisations that still use it they won't break immediately (or at least won't lose the data).

**Comment by christianwach** [comment]
@mattwire Ah yes, good point!

**Comment by eileenmcnaughton** [comment]
@mattwire @mlutfy @seamuslee001  we talked about getting this merged before the next rc is cut  - does anyone feel comfortable enough with this to merge?

**Comment by mattwire** [comment]
Yes, I'm happy with this. I've got it running on a couple of sites. Anyone using multiple languages should do a little extra testing on 5.29 to check everything works ok @kcristiano @aydun @MikeyMJCO @mlutfy
